MetaMask/metamask-extension
# Updating AvatarAccount objects to enums

## Explanation

In an effort to contribute to the TS migration of the extension and to improve DX and consistency when using the DS components this PR converts the AvatarAccount objects to enums. 

`avatar-account.constants.js` => `avatar-account.types.ts`

It follows the following conventions:
- PascalCase
- Singular over Plurals
- Variant over Type

* Fixes #17726 

## Screenshots/Screencaps



### Before

https://user-images.githubusercontent.com/8112138/223860448-4f6540d7-cfa7-4e99-b6c2-b57fd38308bb.mov

### After

https://user-images.githubusercontent.com/8112138/223860538-091e3764-452f-4b85-92e3-572bde6a4382.mov

Check for all old consts in codebase returns no results
![Screenshot 2023-03-08 at 1 38 36 PM](https://user-images.githubusercontent.com/8112138/223860805-f161cdaa-514f-49ec-967c-8c660d3164b4.png)
![Screenshot 2023-03-08 at 1 38 40 PM](https://user-images.githubusercontent.com/8112138/223860807-ead00ff1-845c-4211-9fd5-2345558fbe47.png)
![Screenshot 2023-03-08 at 1 38 47 PM](https://user-images.githubusercontent.com/8112138/223860811-cdb9394b-d17f-4001-aa64-8e07168474a0.png)

Also updated all instances of `AvatarAccount` outside of component folder that use `size` or `variant` prop
![Screenshot 2023-03-08 at 2 09 40 PM](https://user-images.githubusercontent.com/8112138/223862475-a4bf7894-d727-4dd9-9e1e-7376c665545f.png)



## Manual Testing Steps

- Go to the latest build of storybook in this PR 
- Check controls and stories
- Check documentation

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
